src/borrow_tracker/tree_borrows/tree.rs

```diff
@@ -721,9 +721,12 @@ impl<'tcx> Tree {
                     // visit all children, skipping none
                     |_| ContinueTraversal::Recurse,
                     |args: NodeAppArgs<'_>| -> Result<(), TransitionError> {
-                        let NodeAppArgs { node, .. } = args;
+                        let NodeAppArgs { node, perm, .. } = args;
+                        let perm = perm.get().copied().unwrap_or_else(|| node.default_location_state());
                         if global.borrow().protected_tags.get(&node.tag)
                             == Some(&ProtectorKind::StrongProtector)
+                            // TODO Think about what the right semantics should be.  What if only part of the range is Cell.
+                            && !perm.permission().is_cell()
```

Comment:
@RalfJung do you have opinions on what the right semantics should be here? Can you deallocated protected `Cell`s?

Reply:
I would not introduce a special case unless we have a good reason for this. So for now I'd say this should be UB.

Reply:
This would break [this test](https://github.com/rust-lang/miri/blob/master/tests/pass/tree_borrows/interior_mutability.rs#L82).